crates/templates/src/template.rs

```diff
@@ -646,6 +646,7 @@ fn validate_v1_manifest(raw: &RawTemplateManifestV1) -> anyhow::Result<()> {
 mod test {
     use super::*;
 
+    #[allow(dead_code)]
```

Comment:
This one I'm unsure about.  This was added to address the following error:
```
error: field `0` is never read
   --> crates/templates/src/template.rs:649:21
    |
649 |     struct TempFile(tempfile::TempDir, PathBuf);
    |            -------- ^^^^^^^^^^^^^^^^^
    |            |
    |            field in this struct
    |
    = note: `-D dead-code` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(dead_code)]`
help: consider changing the field to be of unit type to suppress this warning while preserving the field numbering, or remove the field
    |
649 |     struct TempFile((), PathBuf);
    |                     ~~

error: could not compile `spin-templates` (lib test) due to 1 previous error
```

However, if I either update as suggested or remove the TempDir field, one of the test starts failing... perhaps due to the temp dir being dropped from memory before the assertion is made?  Better way to address this?

Reply:
Might be able to replace with e.g.
```rust
struct TempFile {
  path: PathBuf,
  _temp_dir: tempfile::TempDir,
}
```

Reply:
Thanks @lann!  Did the trick.